src/python/pants/backend/native/tasks/native_compile.py

```diff
@@ -152,13 +148,19 @@ def _compiler(self):
   def _make_compile_request(self, versioned_target):
     target = versioned_target.target
 
-    include_dirs = [
-      os.path.join(get_buildroot(), dep.address.spec_path)
-      for dep in self.native_deps(target)
-    ] + [
-      os.path.join(get_buildroot(), ext_dep.address.spec_path, ext_dep.include_relpath)
-      for ext_dep in self.packaged_native_deps(target)
-    ]
+    include_dirs = []
+    for dep in self.native_deps(target):
+      source_lib_base_dir = os.path.join(get_buildroot(),
+                                         dep._sources_field.rel_path)
```

Comment:
Honest question: should this be using the `Target.target_base` (ie, the "source root") of the target instead? Not really sure what this should be relative to.

See https://www.pantsbuild.org/setup_repo.html#source-roots for reference.

In the case of synthetic/generated code, or code without an existing source root, a source root is created at the target's definition location (ie, it's `rel_path`/`spec_path`).